dvc/repo/tree.py

```diff
@@ -245,78 +248,117 @@ class RepoTree(BaseTree):  # pylint:disable=abstract-method
     Any kwargs will be passed to `DvcTree()`.
     """
 
-    def __init__(self, repo, **kwargs):
-        super().__init__(repo, {"url": repo.root_dir})
-        if hasattr(repo, "dvc_dir"):
-            self.dvctree = DvcTree(repo, **kwargs)
-        else:
-            # git-only erepo's do not need dvctree
-            self.dvctree = None
+    def __init__(
+        self, tree, subrepos=None, **kwargs
```

Comment:
It looks a bit strange to pass a list of subrepos into RepoTree like this. It seems like a RepoTree should be able to find nested subrepos by walking itself.